qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/JavaDataModelListenerManager.java

```diff
@@ -151,16 +151,47 @@ public boolean visit(IResourceDelta delta) throws CoreException {
 			case IResource.FILE:
 				IFile file = (IFile) resource;
 				if (isJavaFile(file) && isFileContentChanged(delta)) {
-					// A Java file has been saved
-					JavaDataModelChangeEvent event = new JavaDataModelChangeEvent();
-					event.setProjectURIs(new HashSet<String>());
-					event.getProjectURIs().add(JDTQuteProjectUtils.getProjectURI(file.getProject()));
-					fireAsyncEvent(event);
+					// A Java file has been updated and saved

```

Comment:
After some investigation, I wonder if it should be better to use ElementChangedEvent to track the update, delete, and create of Java file. Let me investigate that.

Reply:
@rgrunber do you think it is possible to use JDT ElementChangedEvent to track change of Java file but only on save?

Reply:
It should be possible. The workspace would broadcast a delta for any file that is changed. What do you mean by "save". Do you want to listen to any attempt to save the file, or any attempt to save when the file is dirty ?

Reply:
I would like to avoid sending notification when user is updating the Java editor, but send a notification when user do a save.

Reply:
I am using IResourceChangeListener for now.